http-client-core/src/main/java/io/micronaut/http/client/annotation/Client.java

```diff
@@ -55,6 +57,27 @@
     String id() default "";
 // end::value[]
 
+    /**
+     * The interface definition type. When set to {@link DefinitionType.SERVER} the {@link Produces} and {@link Consumes}
+     * definition evaluated for each method of the interface will be reversed .
+     *
+     * <p>Whilst not necessarily recommended, there are scenarios like testing where it is useful to share a common interface between client and server and use the
+     * interface to create a new client declarative client. The client typically needs to produce the content type accepted by the server and consume the content type
+     * produced by the server. In this arrangement using the interface directly will not result in the correct behaviour.</p>
+     *
+     * <p>In this scenario you can set {@link DefinitionType} to {@link DefinitionType.SERVER} which will ensure the requests sent by the client use the content type declared by the {@link Consumes}
+     *  annotation of the interface and that responses use the content type declared by the {@link Produces}.</p>
+     *
+     * <p>The default behaviour is to use {@link DefinitionType.CLIENT} where the inverse of the above is true.</p>
+     *
+     * @return The interface definition type
+     * @since 4.8.0
+     * @see Consumes
+     * @see Produces
+     */
+    @NonNull
+    DefinitionType definitionType() default DefinitionType.CLIENT;
```

Comment:
can you annotate this with `@Experiemental`

Reply:
fixed